Source/Modules/javascript.cxx

```diff
@@ -103,6 +104,61 @@ class Template {
   String *templateName;
 };
 
+/**
+ * Typescript Type definition emitter helper
+ */
+class TsTypeInterface
+{
+public:
+  enum TsType
+  {
+    interfaceType,
+    enumType
+  };
+
+
+  File *typesFilePtr;
+
+  explicit TsTypeInterface(TsType type) : tsType(type)
+  {
+    functionList = NewHash();
+    variableClassCode = NewString("");
+    functionClassCode = NewString("");
+    tsTypeExtraCode = NewString("");
+    className = NewString("");
```

Comment:
BTW, this seems unnecessary and leaks memory, as we always use `Copy()` below to set it. I.e. I believe it could be just initialized with `NULL` instead (same for `baseClassName` below).